member lookup with Generic base class in 2.5.3

### Steps to reproduce

```python
from abc import ABC
from typing import Generic, TypeVar

Anything = TypeVar("Anything")
MoreSpecific = TypeVar("MoreSpecific", str, int)


class A(ABC, Generic[Anything]):
    def a_method(self) -> None:
        print("hello")


class B(A[MoreSpecific]):
    pass


class C(B[str]):
    pass


c = C()
c.a_method()

```

### Current behavior

pylint fails with `22:0: E1101: Instance of 'C' has no 'a_method' member (no-member)` when using astroid 2.5.3 but doesn't when using astroid 2.5.2

### Expected behavior

### `python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"` output
When failing:
```
> pylint --version
pylint 2.7.4
astroid 2.5.3
Python 3.9.1 (default, Feb 15 2021, 10:36:56)
[Clang 10.0.0 (clang-1000.10.44.4)]
```
Without failing:
```
> pylint --version
pylint 2.7.4
astroid 2.5.2
Python 3.9.1 (default, Feb 15 2021, 10:36:56)
[Clang 10.0.0 (clang-1000.10.44.4)]
```

@hippo91 and @cdce8p: both your test and the `E1136` pass after merging #927 and #946 (I had a suspicion)
```
(venv39) (venv39) ~/p/o/a/fixtures (master) $ cat issue942.py
# pylint: disable=missing-docstring,invalid-name,too-few-public-methods,no-self-use
from abc import ABC
from typing import Generic, TypeVar

Anything = TypeVar("Anything")
MoreSpecific = TypeVar("MoreSpecific", str, int)


class A(ABC, Generic[Anything]):
    def a_method(self) -> None:
        print("hello")


class B(A[MoreSpecific]):
    pass


class C(B[str]):
    pass


c = C()
c.a_method()
(venv39) (venv39) ~/p/o/a/fixtures (master) $ python -m pylint -- issue942.py 
************* Module issue942
issue942.py:23:0: E1101: Instance of 'C' has no 'a_method' member (no-member)

------------------------------------------------------------------
Your code has been rated at 6.15/10 (previous run: 6.15/10, +0.00)

(venv39) (venv39) ~/p/o/a/fixtures (master) $ git co fix/926-strong-mutable-ref-in-context-clone 
Switched to branch 'fix/926-strong-mutable-ref-in-context-clone'
(venv39) (venv39) ~/p/o/a/fixtures (fix/926-strong-mutable-ref-in-context-clone) $ 
python -m pylint -- issue942.py 
************* Module issue942
issue942.py:14:8: E1136: Value 'A' is unsubscriptable (unsubscriptable-object)

------------------------------------------------------------------
Your code has been rated at 6.15/10 (previous run: 6.15/10, +0.00)

(venv39) (venv39) ~/p/o/a/fixtures (fix/926-strong-mutable-ref-in-context-clone) $ 
git co fix/942-merge 
Switched to branch 'fix/942-merge'
(venv39) (venv39) ~/p/o/a/fixtures (fix/942-merge) $ python -m pylint -- issue942.py   # <-- here they pass

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 6.15/10, +3.85)
```

I've pushed my this to https://github.com/nelfin/astroid/tree/fix/942-merge on my fork if you'd like to try something else.

--- a/tests/unittest_scoped_nodes.py
+++ b/tests/unittest_scoped_nodes.py
@@ -2154,5 +2154,31 @@ def test_posonlyargs_default_value():
     assert first_param.value == 1
 
 
+@test_utils.require_version(minver="3.7")
+def test_ancestor_with_generic():
+    # https://github.com/PyCQA/astroid/issues/942
+    tree = builder.parse(
+        """
+    from typing import TypeVar, Generic
+    T = TypeVar("T")
+    class A(Generic[T]):
+        def a_method(self):
+            print("hello")
+    class B(A[T]): pass
+    class C(B[str]): pass
+    """
+    )
+    inferred_b = next(tree["B"].infer())
+    assert [cdef.name for cdef in inferred_b.ancestors()] == ["A", "Generic", "object"]
+
+    inferred_c = next(tree["C"].infer())
+    assert [cdef.name for cdef in inferred_c.ancestors()] == [
+        "B",
+        "A",
+        "Generic",
+        "object",
+    ]
+
+
 if __name__ == "__main__":
     unittest.main()

